### DISCUSSION TITLE
I try to run my fastapi app in docker but I get this error message if i start the container

### DISCUSSION THREAD

**Discussion by liamklaus** [fastapi/fastapi#6977] [Questions] [answered]
database_1  |
database_1  | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1  |
database_1  | 2021-01-21 00:21:49.127 UTC [1] LOG:  starting PostgreSQ
L 12.1 on x86_64-pc-linux-musl, compiled by gcc (Alpine 9.2.0) 9.2.0,64-bit
database_1  | 2021-01-21 00:21:49.135 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1  | 2021-01-21 00:21:49.135 UTC [1] LOG:  listening on IPv6 address "::", port 5432
database_1  | 2021-01-21 00:21:49.157 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
database_1  | 2021-01-21 00:21:49.220 UTC [22] LOG:  database system was shut down at 2021-01-21 00:04:37 UTC
database_1  | 2021-01-21 00:21:49.237 UTC [1] LOG:  database system is ready to accept connections
server_1    | INFO:     Uvicorn running on http://0.0.0.0:8000 (Press CTRL+C to quit)
server_1    | INFO:     Started reloader process [1] using watchgod 
server_1    | Process SpawnProcess-1:
server_1    | Traceback (most recent call last):
server_1    |   File "/usr/local/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
server_1    |     self.run()
server_1    |   File "/usr/local/lib/python3.8/multiprocessing/process.py", line 108, in run
server_1    |     self._target(*self._args, **self._kwargs)
server_1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/subprocess.py", line 61, in subprocess_started
server_1    |     target(sockets=sockets)
server_1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/server.py", line 48, in run
server_1    |     loop.run_until_complete(self.serve(sockets=sockets))

server_1    |   File "uvloop/loop.pyx", line 1456, in uvloop.loop.Loop.run_until_complete
server_1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/server.py", line 55, in serve
server_1    |     config.load()
server_1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/config.py", line 308, in load
server_1    |     self.loaded_app = import_from_string(self.app)
server_1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/importer.py", line 23, in import_from_string
server_1    |     raise exc from None
server_1    |   File "/usr/local/lib/python3.8/site-packages/uvicorn/importer.py", line 20, in import_from_string
server_1    |     module = importlib.import_module(module_str)
server_1    |   File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
server_1    |     return _bootstrap._gcd_import(name[level:], package, level)
server_1    |   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
server_1    |   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
server_1    |   File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
server_1    |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
server_1    |   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
server_1    |   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
server_1    |   File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
server_1    | ModuleNotFoundError: No module named 'app'

### Project structure

       | myproject
            └── src
             |   └── app
             |    |    ├── main.py
             |   └── requirements.txt
            └── tests
            └── venv
            └── docker-compose.yml
 ### Dockfile
`
FROM tiangolo/uvicorn-gunicorn:python3.8

LABEL maintainer="Sebastian Ramirez <tiangolo@gmail.com>"

COPY ./requirements.txt /src/requirements.txt

RUN pip install --no-cache-dir fastapi

COPY . /src/app
`
### Docker-compose
version: '3.7'

services:
  server:
    build:
      context: ./src
      dockerfile: dockerfile
    volumes:
      - ./src/:/src/
      - /var/run/docker.sock:/var/run/docker.sock
    command: uvicorn app.main:app --reload --workers 1 --host 0.0.0.0 --port 8000
    env_file:
      - ./src/app/core/.env
    ports:
    - 8000:8000
    depends_on:
      - database
  database:
    image: postgres:12.1-alpine
    volumes:
    - postgres_data:/var/lib/postgresql/data
    env_file:
      - ./src/app/core/.env
    ports:
    - 5433:5433
volumes:
  postgres_data:

**Comment by krishnardt** [comment]
This line in DocFile has some issue.
COPY . /src/app
Source point is not correctly mentioned.

**Comment by iravivarma** [comment, marked as answer]
If you are in the src folder,
Use just an app instead of a dot.

like 
Copy .src/app 
to 
Copy app /src/app
Here you are copying the app file to the docker from src/app folder
Or else

Once check the COPY Command Format in docker documentation

**Comment by liamklaus** [comment]
Thanks!

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue 👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
